classes/task/ProcessQueueJobs.php

```diff
@@ -38,28 +37,40 @@ public function getName(): string
      */
     public function executeActions(): bool
     {
-        if (Application::isUnderMaintenance() || !Config::getVar('queues', 'job_runner', true)) {
+        // If processing of queue jobs via scheduler is disbaled
+        // will not process any queue jobs via schedule taks
+        if (!Config::getVar('queues', 'schedule_job_process', true)) {
             return true;
         }
 
-        $jobQueue = app('pkpJobQueue');
+        $jobQueue = app('pkpJobQueue'); /** @var \PKP\core\PKPQueueProvider $jobQueue */
 
         $jobBuilder = $jobQueue->getJobModelBuilder();
 
         if ($jobBuilder->count() <= 0) {
             return true;
         }
 
-        // Executes all pending jobs when running the runScheduledTasks.php on the CLI
+        // When processing queue jobs vai schedule task in CLI mode
+        // will process a limited number of jobs at a single time
         if (PKPContainer::getInstance()->runningInConsole('runScheduledTasks.php')) {
-            while ($jobBuilder->count()) {
+            $maxJobCountToProcess = (int)Config::getVar('queues', 'job_runner_max_jobs', 30);
```

Comment:
For the sake of "what if..."

```suggestion
            $maxJobCountToProcess = abs(Config::getVar('queues', 'job_runner_max_jobs', 30));
```